server/server.js

```diff
@@ -140,25 +155,27 @@ function movePeriodic(ctrlls) {
       y = y + height;
     }
     ctrlls[key].y = y;
+    ctrl.angle = angle;
```

Comment:
It's quite confusing that you're referencing the same controller both as `ctrl` and as `ctrlls[key]`.